lib/eclipse/EclipseState/Schedule/TimeMap.cpp

```diff
@@ -189,57 +239,71 @@ namespace Opm {
         }
     }
 
-    void TimeMap::addFromDATESKeyword( const DeckKeyword& DATESKeyword ) {
+    std::time_t TimeMap::timeTFromEclipse(const DeckRecord &dateRecord) {
+        static const std::string errorMsg("The datarecord must consist of the four values "
```

Comment:
I added static `TimeMap::mkdatetime() and simplified the logics down to the following:
```
        const auto &dayItem = dateRecord.getItem(0);
        const auto &monthItem = dateRecord.getItem(1);
        const auto &yearItem = dateRecord.getItem(2);
        const auto &timeItem = dateRecord.getItem(3);

        std::tm tm;
        strptime(timeItem.get<std::string>(0).c_str(), "%H:%M:%S", &tm);

        std::time_t date = mkdatetime(yearItem.get<int>(0),
                                      eclipseMonthIndices().at(monthItem.get<std::string>(0)),
                                      dayItem.get<int>(0),
                                      tm.tm_hour,
                                      tm.tm_min,
                                      tm.tm_sec);
        return date;
```